spec/controllers/answer_tags_controller_spec.rb

```diff
@@ -117,4 +127,46 @@
       end
     end
   end
+
+
+  #To allow creation if not existing and simultaneously updating the new answer tag.
+  #params: answer_id (answer id mapping to which tag is being created)
+  #params: tag_prompt_deployment_id (tag_prompt id mapping to which tag is being created)
+  #params: value (new value to be updated)
+
+  describe '#create_edit' do
+    context 'when student tries to create or update the answer tags' do
+      before(:each) do
+        controller.request.session[:user] = student
+      end
+
+      it 'add entry if not existing and update the old value by new value provided as param' do
+        params = {answer_id: answer.id,tag_prompt_deployment_id: tag_deploy.id,value: 0}
+        post :create_edit, params, session
+        expect(response).to have_http_status(200)
```

Comment:
Rather than just checking whether the response was 200, we also should check whether the value was updated in the database.
```suggestion
        expect(response).to have_http_status(200)
        expect(AnswerTag.find_by(answer_id: answer.id).value).to eql("0")
```